src/codec_aom.c

```diff
@@ -521,6 +522,31 @@ static avifBool avifProcessAOMOptionsPostInit(avifCodec * codec, avifBool alpha)
     return AVIF_TRUE;
 }
 
+struct aomScalingModeMapList
+{
+    avifScalingMode avifMode;
+    AOM_SCALING_MODE aomMode;
+};
+
+static const struct aomScalingModeMapList scalingModeMap[] = {
+    { { 1, 1 }, AOME_NORMAL },    { { 1, 2 }, AOME_ONETWO },    { { 1, 4 }, AOME_ONEFOUR },  { { 1, 8 }, AOME_ONEEIGHT },
+    { { 3, 4 }, AOME_THREEFOUR }, { { 3, 5 }, AOME_THREEFIVE }, { { 4, 5 }, AOME_FOURFIVE },
+};
+
+static const int scalingModeMapSize = sizeof(scalingModeMap) / sizeof(scalingModeMap[0]);
+
+static avifBool avifFindAOMScalingMode(const avifScalingMode * avifMode, AOM_SCALING_MODE * aomMode)
+{
+    for (int i = 0; i < scalingModeMapSize; ++i) {
+        if (scalingModeMap[i].avifMode.numerator == avifMode->numerator && scalingModeMap[i].avifMode.denominator == avifMode->denominator) {
```

Comment:
I realized that we already have a `clapFraction` in `avif.c`. I changed it to `avifFraction` and made it more public, so that I can reuse its simplification ability.